### DISCUSSION TITLE
Scaling quantity ranges for ingredients

### DISCUSSION THREAD

**Discussion by TheElbinho** [mealie-recipes/mealie#2989] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Some recipes give ranges for the quantity of an ingredient like "30-45 grams of gelatin". Currently there is no way to enter this into mealie (somewhat related to https://github.com/mealie-recipes/mealie/issues/2926).

### Please provide a concise description of the feature that would resolve your issue.

Add an operator like "to" to mark a range, so that the input will be accepted and the scaler knows to multiply both ends of the range (so that "30 to 45" scaled by 2 becomes "60 to 90"

### Please consider and list out some caveats or tradeoffs made in your design decision

Could be a translation problem, you probably don't want to change the operator based on the translation, so maybe a dash "-" would be the most universal sign. But dependent on how the quantity input is handled, it would be necessary to make sure that "30-45" doesn't get interpreted as "-15"

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by boc-the-git** [comment]
My suggested solution for this is that in your instance, work out the measurement you want (say 38g in your 30-45 example) and update the recipe to have that.
It gives you a more meaningful recipe and solves this problem.

**Reply by mblauser** [reply]
This would work for recipes that have already been made, likely several times (enough to know the preferred amount).  Mealie seemingly allows for the addition and searching of recipes which may not have been made; certainly not adjusted for the individual taste preference of others.  The author presents a complete (and in my opinion, elegant) enhanced user flow for ingredients.

**Comment by TheElbinho** [comment]
My current workaround is entering the minimum and adding the range in the notes. It is fine, but I think my suggested feature would be better. Although if no-one else sees a problem here, I can live with the featureset as it is

**Reply by mblauser** [reply]
Your suggestion is well thought out from a usability perspective.  A small icon to expand the UI into a range input (by default hidden) would be a nice method to invoke this feature.

**Comment by xad001x0w** [comment]
I'll add my voice to this as something I'd make use of. I've just started using Mealie and the first few recipes I've entered all had ranged ingredients: "3-4 cloves of garlic, 6-7 sage leaves, 100-200ml of water" etc

I'm guessing the simplest option would be to store min and max values for an ingredient if someone enters an ingredient amount including a dash. That way there would still be numeric values in the database to make scaling calculations easier.

**Reply by ChipSkylark37** [reply]
This is half the fun of cooking in not being so strict with the recipe. The recipes you get from others are just their preference. 

This is actually one of the reasons I try to do everything in grams, especially for baking things like pizza dough. Gets away from the '1 cup flour' (packed, loose, heaping?) and say "240g flour" and in the comments "about 1 cup" to reference.

**Comment by ChipSkylark37** [comment]
I  put those variables into the comments section as just 'more or less to taste'. Unless there was a button to the left of the Quantity that could say "about" and then that covers most of the scenario, and/or implementing 'min / max'. But that seems like its geting a lot more complicated. 

When I have certain recipes and want to be as precise as possible, and then can knowingly sway. I know a certain recipe tastes better with a little more vanilla extract, so I'd just adjust the base recipe with the new value and comment it "more or less to taste"

**Comment by henrebotha** [comment]
Related dev comment: https://github.com/mealie-recipes/mealie/issues/1140#issuecomment-1094101232

I think this is an _extremely_ obvious and necessary feature. Loads of recipes use ranges. Saying "the data model doesn't support it" just means the data model is wrong. The data model should be subservient to the use case, not the other way around.
